### ISSUE TITLE
Netsplit can possibly cause Matrix user becoming moderator and deop by server not getting synced?

### ISSUE THREAD

**Issue by Mikaela** [matrix-org/matrix-appservice-irc#518]
https://matrix.to/#/!SudviOJlimDvrGdFCY:matrix.org/%2415099696853688662thvbU:matrix.org

**Comment by Half-Shot** [comment]
I investigated this a while back and totally didn't report it in this bug:

Essentially netsplits seem to upset ChanServ which means if a channel is unlucky enough to lose users in such a way that non-oped people are left, it seems to assign one of them operator powers. This doesn't usually show up on irc clients because the ChanServ is quick to snap back operator to the real users when everything joins up.

I worry that since we burn up on most netsplits on Freenode, we might be missing operator unsets and causing lots of issues with power level syncing.

**Comment by Mikaela** [comment]
> Essentially netsplits seem to upset ChanServ which means if a channel is unlucky enough to lose users in such a way that non-oped people are left, it seems to assign one of them operator powers.

I don't believe ChanServ ever does this and think that the real scenario is that the channel becomes empty and someone /cycles on a splitted server and recreates the channel becoming the op (and Matrix users rejoin at this point?) which then gets reverted when the netsplit is over and the channels get merged to the older one and splitrided modes are unset.

**Comment by Half-Shot** [comment]
>I don't believe ChanServ ever does this

I have logs :). Will append when I do.

**Comment by Mikaela** [comment]
You never uploaded the logs by the way and I agree with my previous comment.

The behaviour you described would be a separate servive known as ChanFix, which is somewhat uncommon.

**Comment by mgttlinger** [comment]
Given that this issue is years old and still users randomly get PL 50 via appservice-irc: Is this being actively investigated?

**Comment by Half-Shot** [comment]
Nope, we're fairly sure what the bug is at this point. It's not easy to fix, though.

**Comment by mgttlinger** [comment]
Okay great that the issue has been located but a bit worrisome that a fix is not in sight given that it is fairly annoying that (mostly) matrix users get assigned PL 50 out of the blue. Is there something I can do to help get this fixed?
